sphinx/util/i18n.py

```diff
@@ -118,3 +118,19 @@ def find_catalog_source_files(locale_dirs, locale, domains=None, gettext_compact
                     catalogs.add(cat)
 
     return catalogs
+
+
+def get_image_filename_for_locale(filename, env, force=False):
```

Comment:
My proposal would be two functions to make the intent clear, one which has the fallback logic and one which just pure, dumb name translation.